### PR TITLE
list/delete custom cross images as well

### PR DESCRIPTION
**Pull request by Emilgardis**

### CODE REVIEW

File: src/bin/commands/images.rs [hunk 1]
```diff
@@ -125,19 +143,27 @@ fn get_cross_images(
...
         .lines()
         .map(parse_image)
         .filter(|image| is_cross_image(&image.repository))
         .filter(|image| local || !is_local_image(&image.tag))
         .collect();
-    images.sort();
 
-    Ok(images)
+    let stdout = cross::docker::subcommand(engine, "images")
+        .args(&["--format", r#"{{.Repository}}:{{.Tag}} {{.ID}}"#])
```
Comment (Alexhuszagh) [inline src/bin/commands/images.rs:157]: I don't think this needs an escape sequence, does it?

Comment (Emilgardis) [reply src/bin/commands/images.rs:160]: oh yeah, that's a remenant of me trying to hack the go formatter to get the labels

sadly its not feasible, you can get them, but I don't know how well supported that is

`{{json .}}` i think was how I got them

File: src/bin/commands/images.rs [hunk 2]
```diff
@@ -178,14 +236,14 @@ pub fn list_images(
 ) -> cross::Result<()> {
     get_cross_images(engine, verbose, true)?
         .iter()
-        .for_each(|line| println!("{}", line.name()));
+        .for_each(|image| println!("{}", image));
```
Comment (Emilgardis) [inline src/bin/commands/images.rs:239]: this fixes an issue with images with only an id

File: src/bin/commands/images.rs [hunk 3]
```diff
@@ -160,16 +185,44 @@ fn rustembedded_target(tag: &str) -> String {
...
+            .cloned()
+        {
+            return Ok(target);
+        }
+    }
+    let mut command = cross::docker::subcommand(engine, "inspect");
+    command.args(&[
+        "--format",
+        &format!(
+            r#"{{index .Config.Labels "{}.for-cross-target"}}"#,
```
Comment (Alexhuszagh) [inline src/bin/commands/images.rs:216]: Does this work if more than one custom image with that label exists, and falls through to here?

Comment (Emilgardis) [reply src/bin/commands/images.rs:216]: oh you're right, an argument is missing

```
.arg(image.id)
```

Comment (Emilgardis) [reply src/bin/commands/images.rs]: I had multiple branches with this change, I picked one that was not fully complete :D, this is now resolved

File: src/bin/commands/images.rs [hunk 4]
```diff
@@ -160,16 +188,46 @@ fn rustembedded_target(tag: &str) -> String {
...
 
-fn get_image_target(image: &Image) -> cross::Result<String> {
-    if let Some(stripped) = image.repository.strip_prefix(GHCR_IO) {
-        Ok(stripped.to_string())
+fn get_image_target(
+    engine: &cross::docker::Engine,
+    image: &Image,
+    target_list: &TargetList,
+) -> cross::Result<String> {
+    if let Some(stripped) = image.repository.strip_prefix(&format!("{GHCR_IO}/")) {
```
Comment (Emilgardis) [inline src/bin/commands/images.rs:196]: without this, targets become `/<target>`, so that they dont match

File: src/bin/commands/images.rs [hunk 5]
```diff
@@ -124,20 +141,31 @@ fn get_cross_images(
...
+        .run_and_get_stdout(verbose)?;
+    let ids: Vec<_> = images.iter().map(|i| i.id.to_string()).collect();
+    images.extend(
+        stdout
+            .lines()
+            .map(parse_image)
+            .filter(|i| !ids.iter().any(|id| id == &i.id))
+            .filter(|image| is_cross_image(&image.repository))
+            .filter(|image| local || !is_local_image(&image.tag)),
+    );
```
Comment (Emilgardis) [inline src/bin/commands/images.rs:166]: skips parsing image meta if they've already been discovered

**Review by Alexhuszagh — APPROVED** [review]

**Pull request merged**